doc/guides/test/ztest.rst

```diff
@@ -377,6 +377,59 @@ Other questions:
   themselves -- only one file to update when changes are made
   eliminates duplication.
 
+Stress test framework
+*********************
+
+Zephyr stress test framework (Ztress) provides environment for executing user
+functions in multiple priority contexts. It can be used to validate that code is
+resilient to preemptions. Framework tracks number of executions and preemptions
+for each context. Execution can have various completion conditions like timeout,
+number of executions or number of preemptions.
+
+Framework is setting up the environment by creating requested number of threads
+(each on different priority), optionally starting a timer. For each context, user
+function (different for each context) is called and then context sleeps for
+randomized amount of system ticks. Framework is tracking CPU load and adjusts sleeping
+periods to achieve higher CPU load. In order to increase probability of preemptions,
+system clock frequency should be relatively high. Default 100 Hz on QEMU x86
+is much too low and it is recommended to increase it to 100 kHz.
+
+Stress test environment is setup and run using :c:macro:`ZTRESS_EXECUTE` which
+accepts variable number of arguments. Each argument is a context which is
+specified by :c:macro:`ZTRESS_TIMER` or :c:macro:`ZTRESS_THREAD` macros. Context
+are specified in priority descending order. Each context specifies completion
+conditions by providing minimum number of executions and preemptions. When all
+conditions are met, run is completed, execution report is printed and macro returns.
+When test is executed, progress report is periodically printed.
+
+Execution can be prematurely completed by specifying test timeout (:c:func:`ztress_set_timeout`)
+or explicit abort (:c:func:`ztress_abort`).
+
+User function parameters contains execution counter and flag indicating if it is
+the last execution.
+
+Example below presents how to setup and run 3 contexts (one is k_timer interrupt handler
+context). Completion criteria is set to at least 10000 executions of each context and
+1000 preemptions of the lowest priority context. Additionally, timeout is configured
+to complete after 10 seconds if those conditions are not met. Last argument of each
+context is initial sleep time which will be adjusted throughout a test to get
+highest CPU load.
+
+  .. code-block:: C
+
+             ztress_set_timeout(K_MSEC(10000));
+             ZTRESS_EXECUTE(ZTRESS_TIMER(foo_0, user_data_0, 10000, Z_TIMEOUT_TICKS(20)),
```

Comment:
If I am understanding things correctly, it looks like the number of preemptions is missing. That is we have four (4) parameters to the ZTRESS_TIMER() macro instead of five (5).

Reply:
Since timer context will always have higher priority than thread it will never be preempted so it only has number of executions.